doc/manual/management_system.rst

```diff
@@ -36,16 +47,14 @@ A controller manager connects to the master and accesses the device database thr
 Git integration
```

Comment:
I think it would be good to go more into the details on how the master-client architecture should work (there are many details which cannot be found in the manual, forum, github, or chat, but which in my opinion are essential to get such setup running). I would also be open to write something up including a diagram and contribute it to the manual if you can help me with the details.

In particular, I mean a setup where there is one computer running `artiq_master` with a bare git repository and at least one client submitting experiments via `artiq_client submit` (or the corresponding python APIs).  My conceptional problem with that setup is that we are missing the nix environment in the bare git repository to actually run `artiq_master`, `artiq_client scan-repository`. I think these details have to be mentioned. A diagram for both setups would also be really helpful.

Reply:
`artiq_master` requires a nix environment, e.g., `nix develop` with a `flake.nix`. However, in a bare git repo there are no checked out files, hence also no `flake.nix`.

Reply:
`flake.nix` has nothing to do with `git` but everything to do with ARTIQ, no?

What is your goal with this section?

From my understanding, you want to suggest to the ARTIQ users a more advanced and powerful way of using ARTIQ and integrating it into a lab setup, correct?
If this is what you want to convey, it's important to give the user the whole story, including every detail to actually run sequences via ARTIQ. If you suggest ARTIQ's client-master architecture together with git repositories but don't mention how to integrate a nix environment, the ARTIQ users won't be able to run sequences in that approach, but then why mention it in the first place?

Reply:
Now, I am maximally confused

Where did you `artiq_*` on the OpenBSD machine without `flake.nix` or conda?

Reply:
Okay, so I assume you intentionally don't want to make the tutorial about nix (flakes) because you consider the particular package manager a user's choice?

Reply:
> I think it would be good to go more into the details on how the master-client architecture should work (there are many details which cannot be found in the manual, forum, github, or chat, but which in my opinion are essential to get such setup running). I would also be open to write something up including a diagram and contribute it to the manual if you can help me with the details.

I'm interested to hear more about what you think is missing, though I suspect it may be material for imminent additions to the dedicated management system page, not so much for this PR. 

> My conceptional problem with that setup is that we are missing the nix environment in the bare git repository to actually run ``artiq_master`` ``artiq_client scan-repository``. I think these details have to be mentioned. A diagram for both setups would also be really helpful.

Various diagrams are being worked on. Details of the Git integration will probably be outlined in at least one, since it's an evident point of confusion.

That said, if you're having issues with running the ARTIQ commands, these are installation problems, and don't have anything to do with the management system setup. The master's Git integration is 100% about using the Git toolset to version control experiment files. In the workflow described in the manual, Git doesn't interact with the management system's organizational files at all, let alone with your `flake.nix`.

Reply:
> Given this context, and considering that the flake.nix file is integral to the project and stored in the project's Git repository, I don't see an "elegant" way to to establish the Nix environment for artiq_master and friends directly from a bare Git repository as it does not contain a working copy of flake.nix.

To be clear, you don't _have_ to store it in the project's Git repository. You _can_, if it's your preferred way to install ARTIQ, and if you'd like the flake.nix to be tracked by Git. You can also store it anywhere else.

Reply:
So maybe something like, having a non-bare mono gitrepo on both client and master computer together with a bare submodule repo inside the `repository` folder is the way to go? What would you recommend?

(Of course one can just `scp` the `flake.nix` to the `artiq_master` computer but I think its extremely valuable to pass on best-practices in the docs - even they may appear opinionated. :) )

Reply:
I can't speak on a recommendation for your workflow, but for the record, the `flake.nix` explicitly doesn't need to be identical on every computer you intend to run some part of the ARTIQ infrastructure on. You can use completely different installation methods and install completely different subsets of the ARTIQ toolset on every machine, and the infrastructure is designed specifically to enable this, e.g. the separation of `artiq-comtools` to allow running controllers and a controller manager without the weight of a full ARTIQ installation. Broadly speaking this is why the manual makes no recommendations about how to version control or communicate your `flake.nix`. There's no standard need to.